executors/keptn/pkg/keptn/keptn.go

```diff
@@ -0,0 +1,329 @@
+		Type:    monitoringType,
+		Project: project,
+		Service: service,
+	}
+
+	source, _ := url.Parse("https://github.com/keptn/keptn/cli#configuremonitoring")
+
+	sdkEvent := cloudevents.NewEvent()
+	sdkEvent.SetID(uuid.New().String())
+	sdkEvent.SetType(keptnlib.ConfigureMonitoringEventType)
+	sdkEvent.SetSource(source.String())
+	sdkEvent.SetDataContentType(cloudevents.ApplicationJSON)
+	sdkEvent.SetData(cloudevents.ApplicationJSON, configureMonitoringEventData)
+
+	eventByte, err := json.Marshal(sdkEvent)
+	if err != nil {
+		return fmt.Errorf("failed to marshal cloud event. %s", err.Error())
+	}
+
+	apiEvent := apimodels.KeptnContextExtendedCE{}
+	err = json.Unmarshal(eventByte, &apiEvent)
+	if err != nil {
+		return fmt.Errorf("failed to map cloud event to API event model. %s", err.Error())
+	}
+
+	_, kErr := k.apiHandler.SendEvent(apiEvent)
+	if err != nil {
+		return fmt.Errorf("sending configure-monitoring event was unsuccessful. %s", *kErr.Message)
+	}
+
+	return nil
+}
+
+func (k *Keptn) GetEvents(service, project, keptnCtx string) error {
+	filter := &apiutils.EventFilter{
+		Project:      project,
+		Service:      service,
+		KeptnContext: keptnCtx,
+		EventType:    "sh.keptn.event.evaluation.finished",
+	}
+
+	eventsCtx, kErr := k.eventHandler.GetEvents(filter)
+	if kErr != nil {
+		return fmt.Errorf("failed to get events for keptn context %s. %s", keptnCtx, *kErr.Message)
+	}
+
+	if len(eventsCtx) != 1 {
+		return fmt.Errorf("expected to see one event of type %s", filter.EventType)
+	}
+
+	if dataMap, ok := eventsCtx[0].Data.(map[string]interface{}); ok {
+		result := dataMap["result"].(string)
+		if result == "pass" {
+			return nil
+		}
+		return ErrEvaluationFailed
+	}
+	return fmt.Errorf("event context data expected to be of type map[string]interface{}")
+}
+
+func (k *Keptn) TriggerEvaluation(service, project, timeframe string) (string, error) {
+	currentTime := time.Now()
+
+	evaluation := apimodels.Evaluation{
+		Start:     currentTime.UTC().Format("2006-01-02T15:04:05"),
+		Timeframe: timeframe,
+	}
+
+	stages, err := k.getProjectStages(project)
+	if err != nil {
+		return "", err
+	}
+
+	stage := stages[0].StageName
+
+	eventCtx, kErr := k.apiHandler.TriggerEvaluation(project, stage, service, evaluation)
+	if kErr != nil {
+		return "", fmt.Errorf("failed to trigger evaluation with err: %v", kErr.GetMessage())
+	}
+	return *eventCtx.KeptnContext, nil
+}
+
+func (k *Keptn) getProjectStages(project string) ([]*apimodels.Stage, error) {
+	proj := apimodels.Project{
+		ProjectName: project,
+	}
+
+	if k.git != nil {
+		proj.GitRemoteURI = k.git.URL
+		proj.GitToken = k.git.Token
+		proj.GitUser = k.git.User
+	}
+
+	p, kErr := k.projectHandler.GetProject(proj)
+	if kErr != nil {
+		return nil, fmt.Errorf("failed to get project with err: %v", kErr.GetMessage())
+	}
+	return p.Stages, nil
+}
+
+type KeptnAPIToken struct {
+	SecretRef *corev1.ObjectReference `json:"secretRef,omitempty"`
+}
+
+type KeptnConfig struct {
```

Comment:
Can these be broken into different files so that we have the keptnconfig.go, since there are functions specific to that struct as well?